python/eups/stack/ProductStack.py

```diff
@@ -690,9 +697,8 @@ def reload(self, flavors=None, persistDir=None, verbose=0):
         for flavor in flavors:
             fileName = self._persistPath(flavor,persistDir)
             self.modtimes[fileName] = os.stat(fileName).st_mtime
-            fd = open(fileName, "rb")
-            lookup = pickle.load(fd)
-            fd.close()
+            with open(fileName, "rb") as fd:
```

Comment:
The code is littered with the old style opens. I have only fixed the one that was in the same file as I was editing.

Reply:
I'm not sure what I'm supposed to do about the race in this code. Maybe it's a bit early in the day for me but that code gets the file name from the flavor, calls stat without checking if the cache file is there, then tries to open it. I think the code is assuming that the `reload` call is gated by a check higher up.